### DISCUSSION TITLE
When to use Show/Switch/Match?

### DISCUSSION THREAD

**Discussion by zhaoyao91** [solidjs/solid#392] [Q&A] [answered]
I found all this could be done by `{xx ? a : b}`

I am not sure what optimization they have and when should I use them instead of raw conditional expresssions.

**Comment by ryansolid** [comment, marked as answer]
Yeah added optimization for the ternary and boolean syntax so they are mostly the same now. You can do keyed modes (where it full replaces children) with the components by having callback children. Switch/Match is a bit cleaner when you have more than 2 outcomes. I do find in general since it avoids `{ }` it makes code cleaner, but that is a preference thing.
